pallets/gas/src/lib.rs

```diff
@@ -314,7 +326,7 @@ pub mod pallet {
                 GasTree::<T>::remove(node_id);
 
                 match node.inner {
-                    ValueType::External { id, value } => {
+                    ValueType::External { id, value } | ValueType::ReservedLocal { id, value } => {
```

Comment:
can we reach `ReservedLocal` nodes in `check_consumed` call?

Reply:
good catch, no we can't, as I can see for now, the `check_consumed` only work for parent nodes, but looks like we can not reach `External` also )